app/helpers/icons.py

```diff
@@ -14,3 +17,9 @@ def get_icon_template_url(base_url='', with_color=True):
         color_part = "-{r},{g},{b}"
     return f"{get_icon_set_template_url(base_url)}/icons/{{icon_name}}" \
            f"@{{icon_scale}}{color_part}.png"
+
+
+def fetch_and_clean_unlisted_sets():
+    unlisted_icon_sets = os.environ.get('UNLISTED_ICON_SETS', '').split(',')
+    cleaned_unlisted_icon_sets = [icon_set for icon_set in unlisted_icon_sets if icon_set]
```

Comment:
Have you tested how this behaves when it receives nothing through the `UNLISTED_ICON_SETS` env variable?
I'm still not quite comfortable with this last `if icon_set` and the default value being an empty string...

Reply:
From my testing it behaves as intended, here's a demonstration : 
![image](https://github.com/user-attachments/assets/6e02d1e8-5f90-47eb-b2e9-993b2f7df296)
(with two unlisted sets)
![image](https://github.com/user-attachments/assets/9ae42908-1423-44bd-aabd-7d6f46f1240c)
(with none)
![image](https://github.com/user-attachments/assets/d210ac3e-465a-42e2-b71f-2fe8a777eab3)
We can also see them absent from sets when the route is pinged.

If I remove the default value, I'm unable to call .split() as the object could be a None (if the key isn't found). This could also trigger KeyError exceptions (https://stackoverflow.com/a/4907053).  We could handle that exception with something like : 
```
try:  
   os.environ["FOO"]
except KeyError: 
   print "Please set the environment variable FOO"
   sys.exit(1)

```
But that would generate added complexity and error management that may not be necessary, if we simply assign a default value and purge it, which seems to be the common way to handle this.

![image](https://github.com/user-attachments/assets/bf9454e8-4bc0-4c10-8832-03487a262934)

Assigning a different default type doesn't work either as we can't call split() on a list, so we would have to cast the list to a string before splitting which is less efficient than the current implementation. 

In regards to `if icon_set`, I'm following [this](https://www.franciscomoretti.com/code-tips/avoid-negative-conditionals) clean coding practice, which from my (limited) experience makes code easier to read. It is also deemed "the more pythonic way" of checking for empty strings in a list (see https://stackoverflow.com/a/3845449)